packages/block-editor/src/components/use-block-sync/index.js

```diff
@@ -84,6 +102,14 @@ export default function useBlockSync( {
 
 	const pendingChanges = useRef( { incoming: null, outgoing: [] } );
 
+	useEffect( () => {
+		setupEntityInstance( entityId, instanceId );
+		return () => removeEntityInstance();
+	}, [ entityId ] );
```

Comment:
It would be a good idea to explain in a comment why `instanceId` isn't in the dependency array here.

Reply:
Since it doesn't matter if it is here or not (instanceId should never change), I went ahead and added it "just in case"